e2e/tests/test_e2e.py

```diff
@@ -0,0 +1,327 @@
+        self.assertEqual(job.spec.schedule, schedule,
+                        "Expected {} schedule, found {}"
+                        .format(schedule, job.spec.schedule))
+
+        # update the cluster-wide image of the logical backup pod
+        image = "test-image-name"
+        config_map_patch = {
+            "data": {
+               "logical_backup_docker_image": image,
+            }
+        }
+        k8s.api.core_v1.patch_namespaced_config_map("postgres-operator", "default", config_map_patch)
+
+        operator_pod = k8s.api.core_v1.list_namespaced_pod(
+            'default', label_selector="name=postgres-operator").items[0].metadata.name
+        k8s.api.core_v1.delete_namespaced_pod(operator_pod, "default")  # restart reloads the conf
+        k8s.wait_for_operator_pod_start()
+
+        jobs = k8s.get_logical_backup_job().items
+        actual_image = jobs[0].spec.job_template.spec.template.spec.containers[0].image
+        self.assertEqual(actual_image, image,
+                        "Expected job image {}, found {}".format(image, actual_image))
+
+        # delete the logical backup cron job
+        pg_patch_disable_backup = {
+            "spec": {
+               "enableLogicalBackup": False,
+            }
+        }
+        k8s.api.custom_objects_api.patch_namespaced_custom_object(
+            "acid.zalan.do", "v1", "default", "postgresqls", "acid-minimal-cluster", pg_patch_disable_backup)
+        k8s.wait_for_logical_backup_job_deletion()
+        jobs = k8s.get_logical_backup_job().items
+        self.assertEqual(0, len(jobs),
+                        "Expected 0 logical backup jobs, found {}".format(len(jobs)))
+
+    def assert_master_is_unique(self, namespace='default', version="acid-minimal-cluster"):
+        """
+           Check that there is a single pod in the k8s cluster with the label "spilo-role=master"
+           To be called manually after operations that affect pods
+        """
+
+        k8s = self.k8s
+        labels = 'spilo-role=master,version=' + version
+
+        num_of_master_pods = k8s.count_pods_with_label(labels, namespace)
+        self.assertEqual(num_of_master_pods, 1, "Expected 1 master pod, found {}".format(num_of_master_pods))
+
+
+class K8sApi:
+
+    def __init__(self):
+
+        # https://github.com/kubernetes-client/python/issues/309
+        warnings.simplefilter("ignore", ResourceWarning)
+
+        self.config = config.load_kube_config()
+        self.k8s_client = client.ApiClient()
+
+        self.core_v1 = client.CoreV1Api()
+        self.apps_v1 = client.AppsV1Api()
+        self.batch_v1_beta1 = client.BatchV1beta1Api()
+        self.custom_objects_api = client.CustomObjectsApi()
+
+
+class K8s:
+    '''
+    Wraps around K8 api client and helper methods.
+    '''
+
+    RETRY_TIMEOUT_SEC = 5
+
+    def __init__(self):
+        self.api = K8sApi()
+
+    def get_spilo_nodes(self, pod_labels, namespace='default'):
```

Comment:
the signature imo would be better with sth like:
```python
def get_nodes_of_pg_cluster(self, cluster_name, cluster_namespace='default')
```
because
1. nodes are not namespaced objects in k8s -> `namespace` parameter alone is a bit confusing
2. `pod_labels` are not general labels but rather the `version` label that by convention in our code base means  the PG cluster name